cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/profiles_populator.py

```diff
@@ -190,39 +176,45 @@ def complete_profile(profile: AzureProfile, profiles: List[AzureProfile]) -> boo
         cli_tell(f"Profile '{profile.name}' information completed: {missing_fields_str}")
         return True
 
-    # filling any information other than just principal_secret requires logging into Azure Account
-    account: Optional[AzureAccountLoginInfo] = None
-    if has_complete_authentication_data(profile):
-        account = azure_login(profile)
-    if account is None:
-        account = cli_azure_login_interactively(profile.subscription_id, profile.tenant_id)
-    if account is None:
-        return False
-    profile.subscription_id = account.subscription_id
-    profile.tenant_id = account.tenant_id
-
-    # fill service principal information
-    if profile.principal_id == "":
-        principal = cli_get_or_create_service_principal(profiles, account.subscription_id)
-        if principal is None:
-            return False
-        profile.principal_id = principal.id
-        profile.principal_secret = principal.secret
-    elif profile.principal_secret == "":
-        profile.principal_secret = find_secret(profile.principal_id, profiles) or cli_ask_secret()
-
-    # fill azure location information
-    if profile.location == "":
-        profile.location = cli_ask_azure_location()
-        if list_invalid_resource_groups(profile):
-            profile.resource_group_names = []
-
-    # fill resource groups information
-    if not profile.resource_group_names:
-        profile.resource_group_names = cli_ask_resource_groups(profile.location)
+    try:
+        # filling any information other than just principal_secret requires logging into Azure Account
+        # login to Azure account
+        if has_complete_authentication_data(profile):
```

Comment:
`has_complete_authentication_data` can be `AzureProfile` method (`has_complete_credentials`).

Reply:
It can, but doesn't need to. Currently I don't see any pros and cons.